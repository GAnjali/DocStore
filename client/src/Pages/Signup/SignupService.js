import { SIGNUP_API_ENDPOINT } from "../../AppConstants";

const axios = require("axios");

const register = async (email, password) => {
  try {
    const response = await axios.post(
      process.env.REACT_APP_SERVER_URL + SIGNUP_API_ENDPOINT,
      {
        email,
        password
      }
    );
    return response;
  } catch (error) {
    return error;
  }
};

export default register;