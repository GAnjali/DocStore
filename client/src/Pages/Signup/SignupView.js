import React from "react";
import {APP_NAME, EMAIL_PLACEHOLDER, PASSWORD_PLACEHOLDER} from "../../AppConstants";

const SignupView = props => {
  return (
    <div className="center">
      <div className="card">
        <h1>{APP_NAME}</h1>
        <form>
          <input
            className="form-item"
            placeholder={ EMAIL_PLACEHOLDER }
            name="email"
            type="text"
            onChange={props.handleChange}
          />
          <input
            className="form-item"
            placeholder={PASSWORD_PLACEHOLDER}
            name="password"
            type="password"
            onChange={props.handleChange}
          />
          {props.hasError && <h6 className={"error"}>{props.error}</h6>}
          <input
            className="form-submit"
            value="SIGNUP"
            type="submit"
            onClick={props.handleFormSignup}
          />
        </form>
      </div>
    </div>
  );
};

export default SignupView;
