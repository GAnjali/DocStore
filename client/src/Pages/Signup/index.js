import React, {Component} from 'react';
import '../Login/Login.css';
import register from "./APIService";
import SignupView from "./SignupView";
import {getToken} from "../../Util/localStorageUtil";

class Index extends Component {

    state = {
        email: '',
        password: '',
        error: '',
        hasError: '',
    };

    componentDidMount() {
        if (getToken().length != 0) {
            this.props.history.replace('/')
        }
    }

    handleChange = (e) => {
        this.setState(
            {
                [e.target.name]: e.target.value
            }
        )
    };

    handleFormSubmit = async (e) => {
        e.preventDefault();
        const response = await register(this.state.username, this.state.password);
        this.handleResponse(response);
    };

    handleResponse = (response) => {
        if (response.status === 201 && response.data.message === "User Added!") {
            this.props.history.replace('/login');
        } else {
            this.setState({
                hasError: true,
                error: response.toString()
            })
        }
    };

    render() {
        return (
            <SignupView email={this.state.email}
                        password={this.state.password}
                        hasError={this.state.hasError}
                        error={this.state.error}
                        handleChange={this.handleChange}
                        handleFormSubmit={this.handleFormSubmit}
            />
        );
    }
}

export default Index;
