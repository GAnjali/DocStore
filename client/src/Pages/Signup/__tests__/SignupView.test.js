import React from "react";
import {shallow} from "enzyme";
import Signup from "../SignupView";

describe("SignupView tests", () => {
    const props = {
        handleChange: jest.fn(),
        handleFormSubmit: jest.fn()
    };
    it("Should render without crashing", () => {
        shallow(<Signup props/>);
    });
    it('should match the signupView snapshot', function () {
        const signupComponent = shallow(<Signup props/>);
        expect(signupComponent).toMatchSnapshot();
    });
});
