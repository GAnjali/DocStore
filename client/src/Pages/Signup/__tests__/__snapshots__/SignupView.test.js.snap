// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupView tests should match the signupView snapshot 1`] = `ShallowWrapper {}`;
